### PR TITLE
Create new Rails App

### PR DESCRIPTION
**Pull request by ZeeshanAzhar23**

### CODE REVIEW

**Comment by ZeeshanAzhar23** [comment]
Implemented the user complete functionality including authentication with devise.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,24 @@
...
+
+* Database initialization
+
+* How to run the test suite
+
+* Services (job queues, cache servers, search engines, etc.)
+
+* Deployment instructions
+
+* ...
```
Comment (javier459) [inline README.md:24]: update the Readme file

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+class UsersController < ApplicationController
+  before_action :find_user, only: [:show,:destroy,:edit,:update]
+  around_action :delete_stripe_customer, only: [:destroy]
+  #before_action :authorize_admin, only: [:index]
```
Comment (javier459) [inline app/controllers/users_controller.rb]: remove commented code

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    authorize @users
+  end
+
+  def show
+
+
+  end
+
+  def edit
+    # @user=User.find(params[:id])
```
Comment (javier459) [inline app/controllers/users_controller.rb]: remove commented code

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
+class UsersController < ApplicationController
+  before_action :find_user, only: [:show,:destroy,:edit,:update]
+  around_action :delete_stripe_customer, only: [:destroy]
+  #before_action :authorize_admin, only: [:index]
+
+
+
+  def index
+    @users= User.all
```
Comment (javier459) [inline app/controllers/users_controller.rb]: spacing issue

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  end
+
+  def edit
+    # @user=User.find(params[:id])
+    authorize @user
+  end
+
+  def update
+    @user=User.find(params[:id])
```
Comment (javier459) [inline app/controllers/users_controller.rb]: we don't need this here as we are finding this in private method below

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  private def find_user
+    @user=User.find(params[:id])
+  end
+
+  private def authorize_admin
+    return unless !current_user.admin?
+    redirect_to root_path, alert: 'Admins only!'
+  end
```
Comment (javier459) [inline app/controllers/users_controller.rb]: don't add private keyword with every method, just add private block and add these functions there.

File: app/helpers/users_helper.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+module UsersHelper
+  def not_found
+    raise ActionController::RoutingError.new('Not Found')
+  end
+  #delete customer from stripe if user choses to delete his acc from app
+  def delete_stripe_customer
+    @cust_id=@user.customer_id
```
Comment (javier459) [inline app/helpers/users_helper.rb]: spacing issue

File: app/models/user.rb [hunk 8]
```diff
@@ -0,0 +1,54 @@
+class User < ApplicationRecord
+  # include HelperMethods
+  # has_secure_password
+  # # attr_accessor :current_password
+  # validate :current_password_is_correct,
+  #          if: :validate_password?, on: :update
```
Comment (javier459) [inline app/models/user.rb]: remove commented code

File: app/models/user.rb [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  validates_confirmation_of :password
+  devise :invitable, :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :subscriptions, dependent: :destroy
+  has_many :plans, through: :subscriptions
+  has_many :subscriptions, dependent: :destroy
+
+  has_one_attached :avatar
+  # after_commit :add_default_avatar, on: %i[update]
```
Comment (javier459) [inline app/models/user.rb]: remove commented code

File: app/models/user.rb [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  #   unless avatar.attached?
+  #     avatar.attach(
+  #       io:File.open(
+  #         Rails.root.join(
+  #         'app', 'assets', 'images', 'default_profile.jpg'
+  #       ), filename: 'default_profile.jpg', content_type: 'image/jpg '
+  #     )
+  #     )
+  #   end
+  # end
```
Comment (javier459) [inline app/models/user.rb]: remove commented code.

**Review by javier459 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ZeeshanAzhar23) [reply README.md:24]: hn g asim bahi wo mn abhi thra bugs fix kr rha us k bd krni update

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (ZeeshanAzhar23) [reply app/controllers/users_controller.rb]: ook ook

**Review by ZeeshanAzhar23 — COMMENTED** [review]
Asim bahi readme ka thra guide kr dn ksa upload krn

**Review by javier459 — APPROVED** [review]

**Pull request merged**